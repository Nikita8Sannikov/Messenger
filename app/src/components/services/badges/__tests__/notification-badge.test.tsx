import { render, screen } from "@testing-library/react";

import NotificationBadge from "@components/services/badges/notification-badge";

describe("NotificationBadge", () => {
	it("renders without content", () => {
		render(<NotificationBadge content={undefined} />);
		const badge = screen.getByTestId("notification-badge");
		expect(badge).toBeInTheDocument();
	});

	it("renders with string content", () => {
		render(<NotificationBadge content="5" />);
		const badge = screen.getByText("5");
		expect(badge).toBeInTheDocument();
	});

	it("renders with number content", () => {
		render(<NotificationBadge content="10" />);
		const badge = screen.getByText("10");
		expect(badge).toBeInTheDocument();
	});

	it("applies custom className", () => {
		render(<NotificationBadge content="3" className="custom-class" />);
		const badge = screen.getByTestId("notification-badge");
		expect(badge).toHaveClass("notification-badge custom-class");
	});

	it("renders with default content value", () => {
		render(<NotificationBadge content={undefined} />);
		const badge = screen.getByTestId("notification-badge");
		expect(badge).toBeInTheDocument();
	});

	it("passes additional props to Badge component", () => {
		const id = "custom-id";
		const { container } = render(<NotificationBadge content="1" id={id} />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with null content", () => {
		const { container } = render(<NotificationBadge content={undefined} />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with string content", () => {
		const { container } = render(<NotificationBadge content="5" />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with number content", () => {
		const { container } = render(<NotificationBadge content="10" />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with custom className", () => {
		const { container } = render(<NotificationBadge content="3" className="custom-class" />);
		expect(container).toMatchSnapshot();
	});
});
