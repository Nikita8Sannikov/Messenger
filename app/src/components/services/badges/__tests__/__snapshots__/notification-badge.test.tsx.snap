// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBadge matches snapshot with custom className 1`] = `
<div>
  <span
    class="MuiBadge-root notification-badge custom-class css-63ezxx-MuiBadge-root"
    data-testid="notification-badge"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
    >
      3
    </span>
  </span>
</div>
`;

exports[`NotificationBadge matches snapshot with null content 1`] = `
<div>
  <span
    class="MuiBadge-root notification-badge css-63ezxx-MuiBadge-root"
    data-testid="notification-badge"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-z28gz6-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`NotificationBadge matches snapshot with number content 1`] = `
<div>
  <span
    class="MuiBadge-root notification-badge css-63ezxx-MuiBadge-root"
    data-testid="notification-badge"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
    >
      10
    </span>
  </span>
</div>
`;

exports[`NotificationBadge matches snapshot with string content 1`] = `
<div>
  <span
    class="MuiBadge-root notification-badge css-63ezxx-MuiBadge-root"
    data-testid="notification-badge"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
    >
      5
    </span>
  </span>
</div>
`;

exports[`NotificationBadge passes additional props to Badge component 1`] = `
<div>
  <span
    class="MuiBadge-root notification-badge css-63ezxx-MuiBadge-root"
    data-testid="notification-badge"
    id="custom-id"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
    >
      1
    </span>
  </span>
</div>
`;
